libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -15,6 +15,8 @@
 
 extern AP_HAL::HAL& hal;
 
+Compass Compass::frontend;
+
```

Comment:
Yeah, I have already went through the implementation. The idea is to have the modules outside the class have \<Class\>::\<function\> interface. Again this interface is not the reason for having static object instead of static pointer to object. @tridge has informed me that InertialSensor should also be converted to have a static object of itself instead of static pointer to object of itself.